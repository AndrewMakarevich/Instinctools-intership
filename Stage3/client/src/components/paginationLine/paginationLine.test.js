import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import PaginationLine from './paginationLine';

const user = userEvent.setup();

describe('Correct pagination line', () => {
  test('render', async () => {
    render(<PaginationLine count={15} page={1} limit={5} setPage={() => {}} />);
    const paginationButtons = await screen.getAllByTestId('pagination-btn');
    expect(screen.getByTestId('pagination-line')).toBeInTheDocument();
    expect(screen.getByTestId('pagination-line')).toMatchSnapshot();
    expect(paginationButtons.length).toBe(5);
  });

  test('buttons click handlers working', async () => {
    const setPage = jest.fn();
    render(<PaginationLine count={15} page={1} limit={5} setPage={setPage} />);
    const paginationButtons = await screen.getAllByTestId('pagination-btn');
    await user.click(paginationButtons[3]);
    expect(setPage.mock.calls[0][0]).toBe(3);
  });
});
