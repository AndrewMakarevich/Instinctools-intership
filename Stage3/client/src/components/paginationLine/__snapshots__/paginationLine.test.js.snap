// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correct pagination line render 1`] = `
<div
  class="pagination-line"
  data-testid="pagination-line"
>
  <div
    class="custom-page-block"
  >
    <input
      class="my-input custom-page-input"
      placeholder="Page number"
      title="Enter your page here"
      type="number"
    />
  </div>
  <button
    class="pagination-btn"
    data-testid="pagination-btn"
    type="button"
  >
    &lt;
  </button>
  <button
    class="pagination-btn active"
    data-testid="pagination-btn"
    type="button"
  >
    1
  </button>
  <button
    class="pagination-btn "
    data-testid="pagination-btn"
    type="button"
  >
    2
  </button>
  <button
    class="pagination-btn "
    data-testid="pagination-btn"
    type="button"
  >
    3
  </button>
  <button
    class="pagination-btn"
    data-testid="pagination-btn"
    type="button"
  >
    &gt;
  </button>
</div>
`;
